src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/DfsController.cs

```diff
@@ -0,0 +1,121 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Core.Modules.Dfs.WebApi.V0.Filter;
+using Catalyst.Core.Modules.Dfs.WebApi.V0.Response;
+using Lib.P2P;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Controllers
+{
+    /// <summary>
+    ///   A base controller for Dfs HTTP API.
+    /// </summary>
+    /// <remarks>
+    ///   Any unhandled exceptions are translated into an <see cref="ApiError"/> by the
+    ///   <see cref="ApiExceptionFilter"/>.
+    /// </remarks>
+    [Route("api/v0")]
+    [Produces("application/json")]
+    [ApiExceptionFilter]
+    public abstract class DfsController : Controller
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Convert this 'abstract' class to a concrete class with a protected constructor.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=5051852)